site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      Message: Route successfully attached to Gateway.
+---
+kind: HTTPRoute
+metadata:
+  name: store-route
+  namespace: store-ns
+spec:
+  parentRefs:
+    - kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: store
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /images
+    backendRefs:
+    - name: image-service
+      port: 8080
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /cart
+    backendRefs:
+    - name: cart-service
+      port: 8080
+status:
+  parents:
+    - parentRef:
+        kind: HTTPRoute
+        name: my-example
+        namespace: prod-gw
+        sectionName: store
+      controllerName: prod-gw
+      conditions:
+      - type: Attached
+        status: true
+        reason: Attached
+        message: Successfully attached to `store` section.
+      grandparents:
+      - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+---
+kind: HTTPRoute
+metadata:
+  name: core-svces
+  namespace: core-svces-ns
+spec:
+  parentRefs:
+  - kind: HTTPRoute
+    name: my-example
+    namespace: prod-gw
+    sectionName: store
+  - kind: HTTPRoute
+    name: my-example
+    namespace: prod-gw
+    sectionName: api
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+    backendRefs:
+    - name: auth-service
+      port: 8080
+  - matches:
+    - path:
+        type: Pathprefix
+        value: /account
+    backendRefs:
+    - name: account-service
+      port: 8080
+status:
+  parents:
+  - parentRef:
+      kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: store
+    controllerName: prod-gw
+    conditions:
+    - type: Attached
+      status: true
+      reason: Attached
+      message: Successfully attached to `store` section.      
+    grandparents:
+    - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+  - parentRef:
+      kind: HTTPRoute
+      name: my-example
+      namespace: prod-gw
+      sectionName: api
+    controllerName: prod-gw
+    conditions:
+    - type: Attached
+      status: true
+      reason: Attached
+      message: Successfully attached to `api` section.
+    grandparents:
+    - parentRef:
+          kind: Gateway
+          name: prod-web-gw
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          Message: Route successfully attached to parent and gateway.
+```
+
+This is pretty verbose, but the extra information will be very important once we
```

Comment:
I would worry about scalability here. If we have many child routes, and change to the parent may require updating 1000s of status resources?

Reply:
Not to mention we are placing all of those dis-aggregated routes into an aggregated status which defeats some of a scalability benefits of decoupling. For the same reasons, Gateway doesn't list all of the attached routes. At large scales it is unsustainable and may not even fit in etcd.

Reply:
Again, the idea here is to avoid the owner of the child Route needing to do a tree traversal to know the state of their child route. I can see the fanout concerns, but was hoping that by having the status refer back _up_ the tree, rather than down, we should avoid some of those problems.